cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -1166,8 +1168,15 @@ buildInplaceUnpackedPackage verbosity
 
         -- Haddock phase
         whenHaddock $
-          annotateFailureNoLog HaddocksFailed $
-          setup haddockCommand haddockFlags []
+          annotateFailureNoLog HaddocksFailed $ do
+            setup haddockCommand haddockFlags []
+            let haddockTarget = elabHaddockForHackage pkg
+            when (haddockTarget == Cabal.ForHackage) $ do
+              let dest = distDirectory </> name <.> "tar.gz"
```

Comment:
I don’t really like the `doc` subdirectory. It already has `docs` in the tarball name and eventually I think it would make sense if the tarballs that need to be uploaded all end up in the same directory.